pubtools/_quay/push_docker.py

```diff
@@ -443,19 +443,25 @@ def remove_old_signatures(
             ) not in new_signatures:
                 signatures_to_remove.append(esig["_id"])
 
-        container_signature_handler.remove_signatures_from_pyxis(signatures_to_remove)
+        if signatures_to_remove:
+            container_signature_handler.remove_signatures_from_pyxis(signatures_to_remove)
 
         signatures_to_remove = []
         if existing_index_images:
             outdated_signatures = []
-            operator_claim_messages = []
-            for item in operator_push_items:
-                operator_claim_messages += (
-                    container_signature_handler.construct_item_claim_messages(item)
+            iib_repo = self.target_settings["quay_operator_repository"]
+            image_schema = "{host}/{repository}:{tag}"
+            new_operator_signatures = []
+            for dest_reg in self.target_settings["docker_settings"]["docker_reference_registry"]:
+                new_operator_signatures.extend(
+                    [
+                        (
+                            eii[0],
+                            image_schema.format(host=dest_reg, repository=iib_repo, tag=eii[1]),
+                        )
+                        for eii in existing_index_images
```

Comment:
existing_index_images are images before iib-add-bundles. Shouldn't new signatures be got from images after iib operation?

Reply:
right, I mixed that up. Fixed